### PR TITLE
Improve Common Crawl download

### LINKED ISSUE #80: [FEA] Improve download and extract utility
Errors when incorrectly using the download and extraction utilities are hard to debug, and the purpose of download and extraction can be unclear.

**Describe the solution you'd like**
For `download_common_crawl` when snapshot numbers are invalid we get this error:
```
Traceback (most recent call last):
  File "/workspace/NeMo-Curator/examples/download_common_crawl.py", line 53, in <module>
    main(attach_args().parse_args())
  File "/workspace/NeMo-Curator/examples/download_common_crawl.py", line 35, in main
    common_crawl = download_common_crawl(
  File "/usr/local/lib/python3.10/dist-packages/nemo_curator/download/commoncrawl.py", line 342, in download_common_crawl
    dataset = download_and_extract(
  File "/usr/local/lib/python3.10/dist-packages/nemo_curator/download/doc_builder.py", line 185, in download_and_extract
    df = dd.from_map(
  File "/usr/local/lib/python3.10/dist-packages/dask/dataframe/io/io.py", line 992, in from_map
    raise ValueError("All `iterables` must have a non-zero length")
```
We should do the following:
- Update the documentation to explicitly call out that the snapshot numbers must be valid
- Provide a [link](https://commoncrawl.org/overview) to the list of valid snapshots in the documentation.
- Catch this error and rethrow an error with a more informative error message.

Furthermore, we should clarify the purpose of the download_and_extract utility. Users should not be under the impression that NeMo Curator does web crawling. We can optionally provide a simple download and extraction utility that operates on raw urls.

### PR DESCRIPTION
**Pull request by ryantwolf**
## Description

Improves the `download_common_crawl` utility by adding more error handling and throwing more descriptive errors.
Closes #80 .

## Usage

Usage is the same as before.
## Checklist

- [x] I am familiar with the [Contributing Guide](https://github.com/NVIDIA/NeMo-Curator/blob/main/CONTRIBUTING.md).
- [x] New or Existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: docs/user-guide/download.rst [hunk 1]
```diff
@@ -39,7 +43,7 @@ By "extraction", we typically mean the process of converting a data format from
     common_crawl = download_common_crawl("/extracted/output/folder", "2020-50", "2021-04", output_type="jsonl")
 
   * ``"/extracted/output/folder"`` is the path to on your local filesystem where the final extracted files will be placed.
-  * ``"2020-50"`` is the first common crawl snapshot that will be included in the download.
+  * ``"2020-50"`` is the first common crawl snapshot that will be included in the download. **Note:** Not every year and week has a snapshot. Ensure that your range includes at least one valid Common Crawl snapshot. A list of valid Common Crawl snapshots can be found `here <https://commoncrawl.org/overview>`_.
```
Comment (sarahyurick) [inline docs/user-guide/download.rst:46]: Thanks for adding a link here, I think it's very useful. I usually refer to https://data.commoncrawl.org/ if maybe we want to add a reference to that somewhere too?

File: tests/test_download.py [hunk 2]
```diff
@@ -7,3 +14,60 @@ def test_imports(self):
         )
 
         assert True
+
+    def test_common_crawl_urls(self):
```
Comment (sarahyurick) [inline tests/test_download.py:18]: I think these tests are very good. I see we're not testing every `raise` that was added by this PR; I don't think that's an issue but just wanted to make sure we don't want them (`raise ValueError("No urls were provided to download")`, etc.).

**Review by sarahyurick — APPROVED** [review]
LGTM! Added very minor comments, but I think this can be merged as is too.

File: docs/user-guide/download.rst (hunk 1 above)
Comment (ryantwolf) [reply docs/user-guide/download.rst]: I like that link a lot more. I changed it

File: tests/test_download.py (hunk 2 above)
Comment (ryantwolf) [reply tests/test_download.py:24]: Good callout. I've added a few more tests.

**Pull request merged**